doc/source/dev/api_guidelines.rst

```diff
@@ -0,0 +1,171 @@
+API Design Guidelines
+=====================
+
+The following section outlines guidelines related to extending and/or modifying
+the Galaxy API. The Galaxy API has grown in an ad-hoc fashion over time by
+many contributors and so clients SHOULD NOT expect the API will conform to
+these guidelines - but developers contributing to the Galaxy API SHOULD follow
+these guidelines.
+
+Before Getting Started
+~~~~~~~~~~~~~~~~~~~~~~
+
+Developers should familiarize themselves with HTTP method definitions 
+https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9. While Galaxy
+follows these imperfectly - a serious effort should be made to use HTTP
+methods (GET, POST, PUT, DELETE, etc.) correctly and consistently. More
+information about these methods with a focus on RESTful APIs can be found
+at https://restfulapi.net/http-methods/.
+
+Developers should familiarize themselves with the HTTP status code definitions
+http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html. The API responses
+should properly set the status code according to the result - in particular
+2XX responses should be used for successful requests, 4XX for various
+kinds of client errors, and 5XX for the errors on the server side.
+
+Developers should also familiarize themselves with RESTful API design
+https://restfulapi.net/rest-api-design-tutorial-with-example/. Again Galaxy
```

Comment:
Minor concern. Who is restfulapi.net? I feel we should use a more transparent source: this one says literally nothing about who the authors are, and their "about us" page, while not providing any information, claims that everything else is "...polluted with duplicate, irrelevant and poor quality information" and presents their site as the source of truth. Even if the content is good, I don't think it justifies using it as a reference in our formal documentation.

There are many good sources. Maybe something like [this](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design) or [this](https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md); there's Roy Fielding's dissertation, etc. Anything, really, just not both anonymous and not peer-reviewed.

Reply:
You're absolutely right - I'll update that link to https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design. The resource looks more complete as well. Thanks!

Reply:
Thank you!